pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -83,58 +83,63 @@ def __init__(
         if not isinstance(sampled_seq, SequenceSamples):
             raise TypeError(
                 "The provided sequence has to be a valid "
-                "pulser.SequenceSamples instance."
+                "SequenceSamples instance."
             )
         if sampled_seq.max_duration == 0:
             raise ValueError("SequenceSamples is empty.")
         # Check compatibility of register and device
-        self.device = device
-        self.device.validate_register(register)
-        self.register = register
+        self.__device = device
+        self.__device.validate_register(register)
+        self.__register = register
```

Comment:
Double leading underscore will mangle the attribute, won't it? Are you sure this is what you want? And if so, why?

Reply:
![image](https://github.com/pasqal-io/Pulser/assets/61453516/001a7e36-99b3-4f13-bb26-7f87be4d99d5)
I went a bit too quick on an article.